🌐 [i18n-KO] Translated `react.md` to Korean

nsbg opened the pull request:
<!-- PR의 제목은 "🌐 [i18n-KO] Translated `<your_file>.md` to Korean" 으로 부탁드립니다 -->
# What does this PR do?

Translated the `react.md` file of the documentation to Korean.
Thank you in advance for your review.

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [x] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

<!-- 1. 위 체크가 모두 완료된 뒤에만 KREW 팀원들에게 리뷰를 요청하는 아래 주석을 노출해주세요!-->
May you please review this PR?
@harheem, @Youngdong2, @xhaktm00, @ssunbear, @ChoHyoungSeo, @Judy-Choi

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

<!-- 2. KREW 팀원들의 리뷰가 끝난 후에 아래 주석을 노출해주세요! -->
@albertvillanova May you please review this PR?

Judy-Choi reviewed (commented):
보금님 꼼꼼하게 살펴보았는데도 모두 정확하고 매끄럽게 번역이 잘 되어서 고칠 내용이 없습니다...! 고생하셨습니다 👍

albertvillanova commented on docs/source/ko/_toctree.yml:
```diff
@@ -2,6 +2,11 @@
   sections:
   - local: index
     title: 소개
+- title: Conceptual guides
+  sections:
+  - local: conceptual_guides/react
+    title: 🤔 멀티스텝 에이전트는 어떻게 동작하나요?
+- title: Examples
```
This is wrongly located: "installation" sub-section below should be located inside "Get started" section.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/smolagents/pr_1666). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

nsbg commented:
Oh I see the Tutorial and Conceptual Guide sections are located incorrectly

nsbg commented:
@albertvillanova I've updated _toctree.yml, ensuring the order matches the original. I've also removed commented-out sections that were not yet translated to keep the content clean.

albertvillanova reviewed (approved):
Thanks!

I have just re-added the commented-out sections in the toctree: because of the way Git wprk, it is easier for new translations to uncomment the new section, rather than having to solve a conflict with the main branch.
